From 04be00943d871a77e70ab4967c7a915bd8d0b2c1 Mon Sep 17 00:00:00 2001
From: Andrew Purtell <apurtell@apache.org>
Date: Thu, 9 Mar 2017 16:54:23 -0800
Subject: [PATCH 338/410] HBASE-17731 Fractional latency reporting in
 MultiThreadedAction

(cherry picked from commit 0bc9ad0628ed96400ae9d0196b3453bfdc48c5b8)

Change-Id: I9961bf092bdf75d7d2d8c60107522c5cc78ebfb1
Author: Andrew Purtell
Reason: Improvement
Ref: CDH-53586
---
 .../hadoop/hbase/util/MultiThreadedAction.java     |    8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/hbase-server/src/test/java/org/apache/hadoop/hbase/util/MultiThreadedAction.java b/hbase-server/src/test/java/org/apache/hadoop/hbase/util/MultiThreadedAction.java
index 5b04ab9..fdea42e 100644
--- a/hbase-server/src/test/java/org/apache/hadoop/hbase/util/MultiThreadedAction.java
+++ b/hbase-server/src/test/java/org/apache/hadoop/hbase/util/MultiThreadedAction.java
@@ -218,11 +218,13 @@ public abstract class MultiThreadedAction {
               + ", time="
               + formatTime(time)
               + ((numKeys > 0 && time > 0) ? (" Overall: [" + "keys/s= "
-                  + numKeys * 1000 / time + ", latency=" + totalOpTime
-                  / numKeys + " ms]") : "")
+                  + numKeys * 1000 / time + ", latency="
+                  + String.format("%.2f", (double)totalOpTime / (double)numKeys)
+                  + " ms]") : "")
               + ((numKeysDelta > 0) ? (" Current: [" + "keys/s="
                   + numKeysDelta * 1000 / REPORTING_INTERVAL_MS + ", latency="
-                  + totalOpTimeDelta / numKeysDelta + " ms]") : "")
+                  + String.format("%.2f", (double)totalOpTimeDelta / (double)numKeysDelta)
+                  + " ms]") : "")
               + progressInfo());
 
           if (streamingCounters) {
-- 
1.7.9.5

